include/aspect/gravity_model/radial.h

```diff
@@ -135,6 +135,52 @@ namespace aspect
         double magnitude_at_bottom;
 
     };
+
+
+
+    template <int dim>
+    class RadialLinearWithTidalPotential : public RadialLinear<dim>
+    {
+      public:
+        /**
+         * Return the gravity vector as a function of position.
+         */
+        Tensor<1,dim> gravity_vector (const Point<dim> &position) const override;
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        void
+        parse_parameters (ParameterHandler &prm) override;
+
+      private:
+        /**
+         * Mass of hosting planet
+         */
+        double M_p;
+
+        /**
+         * Semimajor axis of satellite's orbit
+         */
+        double a_s;
+
+        /**
+         * Rate of non-synchronous rotation in m/year
```

Comment:
Please use SI units wherever possible. You can still allow m/yr as input parameter if the global parameter `Use years instead of seconds` is set, and then convert into m/s when reading and storing the parameter.

And why m as length unit? Wouldnt it be much easier to ask the user to compute the angular velocity themselves beforehand and use that as input? Then you dont need to input the radius of the satellite either, or do you need that for anything else?